lib/plaid.ex

```diff
@@ -88,33 +88,39 @@ defmodule Plaid do
   @spec make_request_with_cred(atom, String.t(), map, map, map, Keyword.t()) ::
           {:ok, HTTPoison.Response.t()} | {:error, HTTPoison.Error.t()}
   def make_request_with_cred(method, endpoint, config, body \\ %{}, headers \\ %{}, options \\ []) do
-    start_time = System.monotonic_time()
-
-    :telemetry.execute(@events_prefix ++ [:start], %{system_time: start_time}, %{
+    common_metadata = %{
       method: method,
       path: endpoint,
       u: :native
-    })
+    }
 
-    try do
-      request_endpoint = "#{get_root_uri(config)}#{endpoint}"
-      cred = Map.delete(config, :root_uri)
-      request_body = Map.merge(body, cred) |> Poison.encode!()
-      request_headers = get_request_headers() |> Map.merge(headers) |> Map.to_list()
-      options = httpoison_request_options() ++ options
+    with_metrics(
```

Comment:
One of the features of instrumentation I had in mind for this library was optionality. Generally I would expect instrumentation to be the responsibility of the application, which might measure Plaid requests alongside other events, such as database calls, using something similar to Appsignal:
```
def make_http_request do
 Appsignal.instrument("plaid-request", fn ->
  Plaid.Transactions.get(...)
 end)
end
```
That said, I think the convenience of out-of-the-box instrumentation for this library is nice feature. I can imagine a scenario where the user has `:telemetry` elsewhere as a dependency in the application, in which case this library could leverage it in `make_request_with_cred/6` in the following way:
```
def make_request_with_cred(method, endpoint, config, body \\ %{}, headers \\ %{}, options \\ []) do
    common_metadata = %{
      method: method,
      path: endpoint,
      u: :native
    }

    request_endpoint = "#{get_root_uri(config)}#{endpoint}"
    cred = Map.delete(config, :root_uri)
    request_body = Map.merge(body, cred) |> Poison.encode!()
    request_headers = get_request_headers() |> Map.merge(headers) |> Map.to_list()
    options = httpoison_request_options() ++ options

    request = fn -> request(method, request_endpoint, request_body, request_headers, options) end

    case Code.ensure_compiled(:telemetry) do
      {:module, _} -> with_metrics(request, common_metadata)
      {:error, _} -> request
    end
  end
```
By making instrumentation optional, users wanting the library just for bindings can keep it lightweight, while those wanting the convenience of standardized instrumentation can take advantage of your feature.

Reply:
Thanks for your thoughts. I'm okay with this implementation given the limitations of the current testing suite.